File: src/kyu_6/pyramid_array/csharp/pyramid_array.cs
using System;

public class Kata {
  public static int[][] Pyramid(int n) {
        int[][] prepared = new int[n][];
        int size = 1;
        for (int i = 0; i < n; i++) {
            prepared[i] = new int[size];
            for (int j = 0; j < size; j++) {
                prepared[i][j] = 1;
            }
            size++;
        }
        return prepared;
  }
}

File: src/kyu_6/pyramid_array/cpp/pyramid_array.cpp
#include <vector>
#include <cstdlib>

std::vector<std::vector<int>> pyramid(std::size_t n) {
  std::vector<std::vector<int>> result;
  for (auto i = 1; i <= n; i ++)
  {
    result.push_back(std::vector<int>(i, 1));
  }
  return result;
}